sdk/client/src/workers/highlight-client-worker.ts

```diff
@@ -143,37 +155,46 @@ function stringifyProperties(
 		const messagesString = stringify({ messages: messages })
 		let payload: PushPayloadMutationVariables = {
 			session_secure_id: sessionSecureID,
+			payload_id: id.toString(),
 			events: { events } as ReplayEventsInput,
 			messages: messagesString,
 			resources: resourcesString,
 			web_socket_events: webSocketEventsString,
 			errors,
 			is_beacon: isBeacon,
 			has_session_unloaded: hasSessionUnloaded,
-			payload_id: id.toString(),
 		}
 		if (highlightLogs) {
 			payload.highlight_logs = highlightLogs
 		}
 
-		const eventsSize = graphqlSDK
-			.PushPayload(payload)
-			.then((res) => res.pushPayload ?? 0)
+		const buf = strToU8(JSON.stringify(payload))
+		const compressed = compressSync(buf, {
+			level: 9,
```

Comment:
should we use the default level / mem here? I think 9 and 12 are the highest values respectively, but in this case because it's only compressed to limit data transmitted I would think this uses too much cpu / mem resources on the client. also curious how this library compares to the browser native https://developer.mozilla.org/en-US/docs/Web/API/CompressionStream

Reply:
my thought here was this is running on a web worker so even if we have higher cpu, we run the risk of blocking the main thread. i'll stick to the defaults tho since there is probably little return for higher background cpu usage.

re. compressionstream - it sounds like [fflate](https://github.com/101arrowz/fflate?tab=readme-ov-file#what-about-compressionstream) thinks that it is better, but i will try some benchmarks and report back

Reply:
tried compressionstream but ran into some issues with getting it to work in the web worker. looks like compressionstream is only supported by browsers of 90% of all devices so i'll stick to fflate for now with the defaults.